### PR TITLE
Restore and Delete Node can be done with multiple ids in one call

### PR DESCRIPTION
**Pull request by lilian-houdelet**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -1154,16 +1154,16 @@ public ResponseEntity<AbstractNode> createNode(@RequestBody AbstractNode node,
...
 
-    @DeleteMapping(value = "/studies/{studyUuid}/tree/nodes/{id}")
+    @DeleteMapping(value = "/studies/{studyUuid}/tree/nodes")
     @Operation(summary = "Delete node with given id")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "the nodes have been successfully deleted"),
         @ApiResponse(responseCode = "404", description = "The study or the node not found")})
     public ResponseEntity<Void> deleteNode(@Parameter(description = "study uuid") @PathVariable("studyUuid") UUID studyUuid,
-                                           @Parameter(description = "id of child to remove") @PathVariable("id") UUID nodeId,
+                                           @Parameter(description = "id of child to remove") @RequestParam("ids") List<UUID> nodeIds,
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1163]: update description

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 2]
```diff
@@ -1597,46 +1597,51 @@ public void restoreNetworkModifications(UUID studyUuid, UUID nodeUuid, List<UUID
...
-            LOGGER.error(e.toString(), e);
-            throw new StudyException(DELETE_NODE_FAILED, e.getMessage());
-        }
 
-        if (startTime.get() != null) {
-            LOGGER.trace("Delete node '{}' of study '{}' : {} seconds", nodeId, studyUuid,
-                    TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime.get()));
-        }
+            if (startTime.get() != null) {
+                LOGGER.trace("Delete node '{}' of study '{}' : {} seconds", nodeId.toString().replaceAll("[\n\r]", "_"), studyUuid,
```
[the remark this answers is not in the document]
Comment (lilian-houdelet) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1638]: I have issue with the log and sonar

[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1638]: what issue ?

[the remark this answers is not in the document]
Comment (lilian-houdelet) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:1638]: don't remember very well, but was about security and I think it was talking about injection,

**Review by souissimai — APPROVED** [review]
it seems good(code and test)

**Pull request merged**